packages/dds/tree/src/simple-tree/api/schemaFactory.ts

```diff
@@ -890,3 +896,72 @@ export function markSchemaMostDerived(schema: TreeNodeSchema): void {
 
 	(schema as typeof TreeNodeValid & TreeNodeSchema).markMostDerived();
 }
+
+/**
+ * Binds metadata to a node schema.
+ * @remarks Accomplishes this by creating a new subclass with the specified metadata, strongly typed.
+ *
+ * @param nodeSchema - The node schema to bind metadata to.
+ * @param metadata - The metadata to bind to the node schema.
+ *
+ * @example
+ *
+ * ```typescript
+ * class Point extends withMetadata(
+ * 	schemaFactory.object("point", { x: schemaFactory.number, y: schemaFactory.number }),
+ * 	{
+ * 		description: "A point in 2D space",
+ * 		custom: {
+ * 			... // Your custom metadata properties here
+ * 		},
+ * 	}
+ * ) {}
```

Comment:
Nit: there's a mix of spaces and tabs in the indentation here, not sure if it'll cause the example to render funky in Intellisense.

Reply:
TSDoc comment lines should always begin with a single space (and I believe the formatter should enforce this). So that at least is expected. I'll double check that there aren't other instances of mixing, though.

Reply:
Yeah, everything looks good as far as I can tell. And intellisense is happy with it.